instance/models/openedx_instance.py

```diff
@@ -409,15 +412,36 @@ def require_user_creation_success(self):
 
     def terminate_obsolete_appservers(self, days=2):
         """
-        Terminate app servers that were created (more than) `days`
-        before the currently-active app server of this instance.
-
-        Do nothing if this instance doesn't have an active app server.
+        Terminate app servers that were created more than `days` before now, except:
+        - the active appserver if there is one,
+        - a release candidate (rc) appserver, to allow testing before the next appserver activation
+          (we keep the most recent running appserver)
+        - a fallback appserver, for `days` after activating an appserver, to allow reverts
+          (we keep the most recent running appserver created before the current activation)
         """
         active_appserver = self.active_appserver
-        if active_appserver:
-            for appserver in self.appserver_set.all():
-                if sufficient_time_passed(appserver.created, active_appserver.created, days):
+        fallback_appserver = None
+        rc_appserver = None
+        now = timezone.now()
+
+        for appserver in self.appserver_set.all().order_by('-created'):
+            if appserver == active_appserver:
+                continue
+
+            # Keep a running appserver as fallback for `days` after activation, to allow reverts
+            if active_appserver and appserver.created < active_appserver.last_activated:
+                if not sufficient_time_passed(active_appserver.last_activated, now, days) \
```

Comment:
@antoviaque If there is a negative delta between the first and the second date passed to `sufficient_time_passed` (i.e., if the first date is more recent than the second one), the function will return `False`. If that situation ever came up (highly unlikely), `not sufficient_time_passed(...)` would evaluate to `True` and we'd end up keeping an additional `fallback_appserver`.

Just wanted to mention this; no need to change anything. The "dangerous" code that terminates VMs below will only run if `sufficient_time_passed` returns `True`, so a negative delta between the first and second date would not cause VMs to get terminated.